test/singleton.js

```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+var S = require('..');
+var eq = require('./internal/eq');
```

Comment:
I've one last nit to pick. I like to group imports. For most modules there are up to three import groups:

  - standard library imports (e.g. `fs`);
  - third-party imports (e.g. `ramda`); and
  - relative imports (e.g. `./models/User`).

In the case of tests there's a fourth possible import group, for the module being tested (e.g. `..`). I like to insert this group between third-party imports and relative imports.

Please add an empty line between the imports above. :)